src/sync/planner.rs

```diff
@@ -48,6 +48,31 @@ impl SeafowlSyncPlanner {
         }
     }
 
+    pub(super) async fn plan_iceberg_syncs(
+        &self,
+        syncs: &[DataSyncItem],
+        table_schema: Arc<arrow_schema::Schema>,
+        table_provider: Arc<dyn TableProvider>,
+    ) -> SyncResult<Arc<dyn ExecutionPlan>> {
+        // Convert the custom Iceberg table provider into a base logical plan
+        let base_plan = LogicalPlanBuilder::scan(
+            LOWER_REL,
+            provider_as_source(table_provider),
+            None,
+        )?
+        .build()?;
```

Comment:
The base scan/plan should only touch stuff that we're changing. In case of Iceberg we support only INSERTs for the time being, and so the base plan should be empty. Otherwise I think this will lead to data duplication on subsequent invocations, since we take everything from the current table state add the new stuff on top of it and dump that to a new table state (where we also inherit the manifests from the previous state)?

Also we might want to do some validation on batches having no UPDATES/DELETES, as those won't be correctly represented?